### DISCUSSION TITLE
Download link to ZIP not available anymore

### DISCUSSION THREAD

**Discussion by christianTF** [community/community#79047] [Repositories]
### Select Topic Area

Bug

### Body

The link to download the ZIP of the code is not available anymore. 
If you right-click the "Download ZIP", you cannot get the direct download url of the zip anymore. 
A download should have a download link.

![image](https://github.com/community/community/assets/14220265/8df07512-8f1a-4cd7-bb88-681ebfbcd76e)

We use the link as we publish the link to beta users, and also use it in our packages for automatic updates.

Why can't a download link simply stay a download link?

Thanks
Christian

**Comment by KavindaBandara321** [comment]
GitHub's UI allows users to download a ZIP file of a repository by clicking the "Code" button and selecting "Download ZIP." The direct download link for the ZIP file is generated dynamically by GitHub and is not immediately visible in the web interface.

**Reply by Baertram** [reply]
And what about non github users who jus want to get the files, and are not comfortable or know how to click where, to get them?
What's the matter with simply keeping things they are, instead of trying to change everyting to be visually perfect but functionally bugs :-(

**Comment by christianTF** [comment]
Yes, this is the reason for my post. Because this was possible since some days or weeks, and now the Download link is no link anymore.
I don’t see the advantage of the invisible link, it is unnecessarily a non-Standard-html foolishness.

**Comment by Tatsh** [comment]
I have this exact issue wanting to paste links to others and inside Gentoo ebuilds (and MacPorts ports) as source URIs. Why is the URL being obscured?

**Reply by eli-schwartz** [reply]
You should not be using these links inside gentoo ebuilds, because the link that has turned into javascript is the branch link -- which is explicitly not stable, it will change the contents as new commits are pushed to branches.

The Tags / Releases page has not changed the links there, which continue to work fine and are clickable / copy-pasteable.

**Reply by Tatsh** [reply]
I only use it on ebuilds that need to use a SHA. I do not link to branches. I would normally go to the SHA listing and right-click _Download ZIP_ for a direct link (and turn it to `tar.gz`).

https://github.com/Tatsh/tatsh-overlay/blob/master/games-emulation/cemu/cemu-2.0_p20231210.ebuild#L13

**Comment by Baertram** [comment]
I ave te same problem. It was working that way before and where is the problem with keeping it as it was? Hiding the links is just a nonsense visual whatever b**ls**t :-( No security or any other reason that I could think off.

Removing the direct download link will make it hard for non GitHub users to get the needed files, and we do not want to create "Release" for small scripts etc.

Please add the direct possibility to copy the download link again, like it was working for years! Thank you very much

**Comment by christianTF** [comment]
I just wanted to re-mention, that the ZIP download button is b*llsh*t, as I now need to download the zip, upload it again somewhere, to provide a download link to the update procedure of the software on the Raspberry.

Please remove UI code control from playing children back to serious developers (I know kiddies it is funny to play with javascript but we need to work).
 
We simply don't want to do a *** release for a test installation, but only get the download link! Thanks to HTML this was and should be very easy.

**Reply by eli-schwartz** [reply]
Working around this UX weirdness doesn't require anything as excessive as downloading it and then reuploading it somewhere. The download link still exists but is simply not advertised... So you can simply add to your readme: "GitHub has started to hide the download link, but it can be found at https://github.com/user/repo/archive/master.zip"

**Reply by Baertram** [reply]
Yeah and I always have to check if the link is correct that I manually wrote there as branches and zip names differ some time..
This is annoying and just a downgrade in usability.

Another workaround would be download the zip and then check the downloads of the browser, and copy the link from there (as most browsers show it then). But these all are workarounds for a feature NO ONE ever asked for. It's just "visually improving something that noone needs". Should simply not have changed, or is there any reason in terms of security etc? I cannot find any.

**Comment by ct0br0** [comment]
I'd love to know what benefit github has by axing this. Justify it.

**Comment by mschlenstedt** [comment]
Well, maybe this (java)scriptkiddie was fired... Anyhow, they removed this bllsht javascript hiding whatever link and re-estableshed the simple and working HTML download link again.

THANK YOU!

**Comment by kuixiong** [comment]
Spent a lot of time trying to figure out why it does not work like it did, and asked the stupid ChatGPT many times, all it told me was copying the link from the button which is obviously not working on may laptop. Now finally get here and see why it is not working 😢

**Reply by Baertram** [reply]
It should be working fine since some weeks again, they reverted the stupid change and made a right click -> copy link possible again.
At least in Firefox browser it works.
